### PR TITLE
Base avatar collisions on the dimensions of the avatar model, add flying mode.

### PR DESCRIPTION
**Pull request by Gigabyte5671**
Collisions:
- Base avatar collisions on the dimensions of the avatar model.
- Move the root point of the avatar to the bottom of the mesh, instead of the center.
- Adjust the default walk/run speed to better match the animation speed and reduce the risk of clipping through terrain.
  - Walk reduced from `4` units to `2.4`.
  - Run reduced from `10` units to `5.2`.
- Implement flying mode.
  - Press `F` (default) to toggle fly mode.
  - Move horizontally (relative to the camera angle) with `WASD` (default).
  - Move vertically up with `Space` (default) and down with `C` (default).

Other:
- Prevent nested keys from being erroneously removed from the Store when saved to persistent storage. This was causing unpredictable behaviour when reloading the client.
- Bundle the avatar animations model with the Web App for ease of development.

### CODE REVIEW

File: src/modules/avatar/controller/inputController.ts [hunk 1]
```diff
@@ -269,6 +274,15 @@ export class InputController extends ScriptComponent {
         return this._avatarState.landSpeed;
     }
 
+    @inspectorAccessor({ min: 0.1, max: 20 })
+    public set flySpeed(value: number) {
+        this._avatarState.flySpeed = value;
+    }
+
+    public get flySpeed() : number {
```
Comment (ctrlaltdavid) [inline src/modules/avatar/controller/inputController.ts:282]:
```suggestion
    public get flySpeed(): number {
```

File: src/modules/avatar/controller/inputController.ts [hunk 2]
```diff
@@ -510,8 +545,10 @@ export class InputController extends ScriptComponent {
                 // Allow the avatar to move in the air.
                 this._doMoveInJumping(delta);
                 // Move to the nxt jump substate once the avatar touches the ground.
-                if (this._detectGround()
-                    || Math.abs(this._avatarState.previousPosY - this._gameObject.position.y) < 0.001) {
+                if (
+                    this._detectGround()
+                    || Math.abs(this._avatarState.previousPosY - this._gameObject.position.y) < 0.001
+                ) {
```
Comment (ctrlaltdavid) [inline src/modules/avatar/controller/inputController.ts:551]: Looks a bit weird but if the linter lets it past, OK.

File: src/modules/avatar/controller/inputController.ts [hunk 3]
```diff
@@ -530,18 +567,46 @@ export class InputController extends ScriptComponent {
...
+        // Rotate the avatar relative to the yaw direction of the camera.
+        const yawRotation = this._defaultCameraAlpha - this._camera.alpha;
+        // Rotate the avatar relative to the pitch direction of the camera.
+        const pitchRotation = this._camera.beta - this._defaultCameraBeta;
+        Quaternion.FromEulerAnglesToRef(pitchRotation, yawRotation, 0, this._gameObject.rotationQuaternion as Quaternion);
+
+        if (this._avatarState.moveDir.x !== 0 || this._avatarState.moveDir.y !== 0 || this._avatarState.moveDir.z !== 0) {
+            this._getCurrentSpeed();
+            // Move on the X-Z plane, relative to the camera angle.
+            const velcoity = this._gameObject
```
Comment (ctrlaltdavid) [inline src/modules/avatar/controller/inputController.ts:591]:
```suggestion
            const velocity = this._gameObject
```

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

File: src/modules/avatar/controller/inputController.ts (hunk 2 above)
Comment (Gigabyte5671) [reply src/modules/avatar/controller/inputController.ts]: Good spotting. This statement exceeded the line length limit at some point in the past. I didn't think to check if it was still too long before reformatting it.
Updated with commit [c245b5b](https://github.com/vircadia/vircadia-web/pull/109/commits/c245b5b858911508e6c81a1d0d1d8ba448b9d9b9), thanks!

**Review by ctrlaltdavid — APPROVED** [review]

**Pull request merged**